sockeye/constants.py

```diff
@@ -322,14 +323,24 @@
 OUTPUT_HANDLER_ALIGN_PLOT = "align_plot"
 OUTPUT_HANDLER_ALIGN_TEXT = "align_text"
 OUTPUT_HANDLER_BEAM_STORE = "beam_store"
+OUTPUT_HANDLER_NBEST = "nbest_translation"
+OUTPUT_HANDLER_NBEST_WITH_SCORE = "nbest_translation_with_score"
+OUTPUT_HANDLER_NBEST_WITH_ALIGNMENTS = "nbest_translation_with_alignments"
 OUTPUT_HANDLERS = [OUTPUT_HANDLER_TRANSLATION,
                    OUTPUT_HANDLER_TRANSLATION_WITH_SCORE,
                    OUTPUT_HANDLER_TRANSLATION_WITH_ALIGNMENTS,
                    OUTPUT_HANDLER_TRANSLATION_WITH_ALIGNMENT_MATRIX,
                    OUTPUT_HANDLER_BENCHMARK,
                    OUTPUT_HANDLER_ALIGN_PLOT,
                    OUTPUT_HANDLER_ALIGN_TEXT,
-                   OUTPUT_HANDLER_BEAM_STORE]
+                   OUTPUT_HANDLER_BEAM_STORE,
+                   OUTPUT_HANDLER_NBEST,
+                   OUTPUT_HANDLER_NBEST_WITH_SCORE,
+                   OUTPUT_HANDLER_NBEST_WITH_ALIGNMENTS]
+
+OUTPUT_HANDLERS_NBEST = [OUTPUT_HANDLER_NBEST,
+                         OUTPUT_HANDLER_NBEST_WITH_ALIGNMENTS,
+                         OUTPUT_HANDLER_NBEST_WITH_SCORE]
```

Comment:
Instead of expanding the set of output handlers and exposing them as independent choices to the command-line CLI, wouldn't it make sense to just apply the existing output handlers to every item on the n-best list when `--nbest-size > 1`? This seems redundant here.

Reply:
Thanks Matt! The question is what behaviour is expected by users. So, you would for instance use `StringOutputHandler` for nbest translation, let's say with `--nbest-size 5` and silently output 5 times the amount of lines?

Reply:
I think that makes more sense to me: the chosen output format gets applied to all items. Then you could add a 'json' output handler that would do different things. It seems convoluted to me to have different output handlers that are all outputting different fields in a JSON object; instead, just return a JSON object with everything, and users can ignore the fields they don't want.

Reply:
Ok, I could live with that, but still think you could, by the same logic, say that an output handler for 1-best translations that outputs the strings without the scores is redundant. @fhieber @tdomhan Your take?

Reply:
I see your point, but I also see a difference between outputting (a) plain text (which is what the existing output handlers moderate) and (b) a structured object like JSON. For the latter, you might as well just include everything that is free to compute, since you have to parse the object anyway.

Reply:
I would keep the basic contract for the translate CLI that n lines in mean n lines out. This clearly means that nbest_size > 1 should return some structured object that holds multiple hypotheses.
Regarding the contents of that object: the additional data (alignments & scores) are computed anyway so there is no additional cost of adding them to the JSON so the 3 different nbest output handlers seem redundant to me.

Reply:
Ok, will remove all handlers except one, which always outputs all fields.